cli/tests/integration_tests.rs

```diff
@@ -344,6 +345,33 @@ mod integration {
     assert_eq!(out, "");
     // TODO(ry) Is there some way to check that the file was actually cached in
     // DENO_DIR?
+
+    let prg = util::deno_exe_path();
+    let output = Command::new(prg)
+      .env("DENO_DIR", deno_dir.path())
+      .env("NO_COLOR", "1")
+      .current_dir(util::root_path())
+      .arg("info")
+      .arg(module_url.to_string())
+      .output()
+      .expect("Failed to spawn script");
+
+    let str_output = std::str::from_utf8(&output.stdout).unwrap();
+    let max = str_output.find("type").unwrap();
+    let cached_file = std::path::PathBuf::from(&str_output[7..max].trim());
+    assert!(cached_file.is_file());
+
+    let module_original = reqwest::get(module_url)
+      .await
+      .unwrap()
+      .text()
+      .await
+      .unwrap();
+    let mut module_cached = String::new();
+    let mut module_file = fs::File::open(cached_file).unwrap();
+    module_file.read_to_string(&mut module_cached).unwrap();
+
+    assert_eq!(module_original, module_cached);
```

Comment:
I think this only partially checks the feature of `deno cache` because `cli/tests/006_url_imports.ts` has dependencies `./cli/tests/subdir/mod2.ts` and `./cli/tests/subdir/print_hello.ts` but these are not checked. `deno cache` also caches the compiled versions of the scripts. That is not checked either, and probably a little hard to check.

Deno has HTTP_PROXY feature and I think it can be used for simulating no-internet environtment. How about running the above script (http://localhost:4545/cli/tests/006_url_imports.ts) with invalid HTTP_PROXY value? If that runs without error, that should check that everything is cached.

Reply:
I agree. Ideally can check for dependencies as well. But since we cannot do it programmatically, I guess it's safe to check it manually?

And for checking the compiled version, I think what we can do now is to try running the compiled js in `gen` folder, and assert the output is as expected. Sounds good?

Reply:
> And for checking the compiled version, I think what we can do now is to try running the compiled js in gen folder, and assert the output is as expected.

That's one approach. But I would recommend using `HTTP_PROXY` env var like the below:

```rust
let output = Command::new(util::deno_exe_path())
  .env("DENO_DIR", deno_dir.path())
  .env("HTTP_PROXY", "http://not-existing-server/")
  .current_dir(util::root_path())
  .arg("run")
  .arg(module_url.to_string())
  .output()
  .expect("Failed to spawn script");
```
With HTTP_PROXY set to not-existing-server this deno process can't access to the script hosted on the server, but it still can execute it. I think that proves the entire dependency is cached in the disk.

Reply:
I see. Then that would mean we don't have to check the file itself because successful run = successful cache?

Reply:
> Then that would mean we don't have to check the file itself because successful run = successful cache?

Yes, I think so.